### PR TITLE
Resubmit expired jobs to make plots using new NMMA flag

### PR DESCRIPTION
**Pull request by bfhealy**
This PR proposes two pseudocode ways to resubmit expired analysis jobs using the `--skip-sampling` flag (https://github.com/nuclear-multimessenger-astronomy/nmma/pull/266) to generate plots from the latest checkpoint results.

Option A: modify submission code to submit both `pending` and `expired` jobs. If a job is marked as `expired`, use the `--skip-sapmling` option in NMMA analysis. Afterward, change each `expired` job to some other status.

Option B: modify retrieval code to submit expired jobs immediately once they are identified.

In each case, we might not want to delete `expired` jobs from the database if we're fetching results plots from them.

### CODE REVIEW

**Comment by Theodlz** [comment]
Hi @bfhealy, maybe you can take a look at the changes I made based on your suggestions and some discussions at the latest SkyPortal weekly meeting? Here's a summary:
1. Now, a job expiration date isn't set by SkyPortal, but by ourselves, in the config. We still enforce the SP webhook's expirated_at, but we make sure that the timeout value set on SkyPortal is larger than the one set in our config.
2. The default job expiration date is set to 6 hours.
3. If a job expires, the retrieval queue sets the analysis status to "job_expired" and we cancel it on expanse.
4. The submission queue retrieves both pending and job_expired analyses and submits the slurm script with the right arguments to either run an analysis or just generate the plots. Then, it sets the status to "running_plot"
5. Both running and running_plot analyses are retrieved from the DB here, and the results uploaded back to SkyPortal once ready. 
6. It's an edge case, but if the job to generate the only the plots also timeouts, we cancel the job and upload a failure back to SkyPortal.


Let me know if you are happy with these changes, and we can merge!

File: nmma_api/services/retrieval_queue.py
Comment (bfhealy) [inline nmma_api/services/retrieval_queue.py]: expanse instead of expansion?

File: nmma_api/services/retrieval_queue.py
Comment (bfhealy) [inline nmma_api/services/retrieval_queue.py]: I think we can remove my import of `submit` here along with the noqa comment, since we ended up not using the submit code in this service.

**Review by bfhealy — COMMENTED** [review]
Hi @Theodlz, thanks very much for making these changes! They all sound good to me. Just a couple minor comments before merging.

File: nmma_api/services/retrieval_queue.py
Comment (Theodlz) [reply nmma_api/services/retrieval_queue.py]: oups :) Copilot bamboozled me. Thanks for checking!

**Review by Theodlz — COMMENTED** [review]
LGTM 🚀

**Review by Theodlz — APPROVED** [review]
LGTM again! (the merge button is stuck yikes, trying to fix it somehow)

**Comment by Theodlz** [comment]
Ended up doing it manually, but the GitHub actions aren't running... Let's wait and see if GitHub goes back to normal soon.

EDIT: manually deployed as well.

**Comment by Theodlz** [comment]
@bfhealy do we need to update the version of NMMA that's on expanse for this to work properly?

**Comment by bfhealy** [comment]
@Theodlz I think I've pushed all the necessary changes to NMMA (and updated the slurm script on Expanse) to make this work.

**Pull request merged**
